### PR TITLE
chore: move generated resources to vault package

### PR DESCRIPTION
**Pull request by fairclothjm**
### Description

All resources under the “generated” directory need to be moved to the vault package to solve an issue with circular dependencies that occurs when moving to a multiplexed provider. These "generated" resources are no longer being maintained by the generation tooling but are under manual maintenance now.

I tried to maintain the code exactly as-is except for any required renamed variables to avoid collision in the vault package. This means I intentionally did not move to use the CreateContext, etc functions. That work is out of scope for this task.

### CODE REVIEW

File: vault/provider.go [hunk 1]
```diff
@@ -173,6 +173,14 @@ var (
 			Resource:      UpdateSchemaResource(pkiSecretBackendKeysDataSource()),
 			PathInventory: []string{"/pki/keys"},
 		},
+		"vault_transform_encode": {
```
Comment (vinay-gopalan) [inline vault/provider.go:176]: Looks like the tests are expecting an `encode_role` resource. We can either update this or update the tests, I think either should be fine :) 
```suggestion
		"vault_transform_encode_role": {
```

File: vault/resource_transform_alphabet_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAlphabetName(t *testing.T) {
```
Comment (vinay-gopalan) [inline vault/resource_transform_alphabet_test.go:18]: nit: should we update the test names to match other tests in repo?
```suggestion
func TestAccTransformAlphabetName(t *testing.T) {
```

**Review by vinay-gopalan — APPROVED** [review]
Looking great! I think we need to update for 1 failing test, and had a suggestion on updating test names to match with other resources within the repo. Good to go otherwise! 👍🏼

File: vault/provider.go (hunk 1 above)
Comment (fairclothjm) [reply vault/provider.go:176]: Thanks! There was inconsistency here across the tests/docs and actual implementation. But the resource is [registered](https://github.com/hashicorp/terraform-provider-vault/blob/bff5e91e7289fc847e4fb4c2e2044b78ffb2aa76/generated/terraform_registry.go#L20) as `vault_transform_decode` so I will stick with that.

File: vault/resource_transform_alphabet_test.go (hunk 2 above)
Comment (fairclothjm) [reply vault/resource_transform_alphabet_test.go:18]: Woops, thanks!

**Pull request merged**